src/AutoRest.CSharp/LowLevel/Generation/DpgClientWriter.cs

```diff
@@ -107,7 +107,7 @@ public void WriteClient()
 
         private void WriteLongRunningResultRetrievalMethods()
         {
-            foreach (var method in _client.ClientMethods.Select(c => c.LongRunningResultRetrievalMethod).WhereNotNull())
+            foreach (var method in _client.ClientMethods.Select(c => c.LongRunningResultRetrievalMethod).WhereNotNull().Distinct())
```

Comment:
Distinct will distinct according to the reference. Are all the LongRunningResultRetrievalMethod in ClientMethods the same instance?

Reply:
> Distinct will distinct according to the reference. Are all the LongRunningResultRetrievalMethod in ClientMethods the same instance?

For record, it's not. See: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record#value-equality

> For types with the record modifier (record class, record struct, and readonly record struct), two objects are equal if they are of the same type and store the same values.

Reply:
Good to known. thanks.